### PR TITLE
Add some useful comments for hot region scheduler.

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?
Hot region scheduler is lack of comment and hard to understand, this request add some useful comments for it.

### CODE REVIEW

File: server/statistics/region_stat_informer.go [hunk 1]
```diff
@@ -18,7 +18,9 @@ import "github.com/pingcap/pd/v4/server/core"
 // RegionStatInformer provides access to a shared informer of statistics.
 type RegionStatInformer interface {
 	IsRegionHot(region *core.RegionInfo) bool
+	// RegionReadStats return the storeID -> write stat of peers on this store
```
Comment (rleungx) [inline server/statistics/region_stat_informer.go:21]:
```suggestion
	// RegionWriteStats return the storeID -> write stat of peers on this store
```

File: server/schedulers/hot_region.go [hunk 2]
```diff
@@ -95,11 +95,13 @@ type hotScheduler struct {
 	r           *rand.Rand
 
 	// states across multiple `Schedule` calls
-	pendings       [resourceTypeLen]map[*pendingInfluence]struct{}
+	pendings [resourceTypeLen]map[*pendingInfluence]struct{}
+	// regionPendings stores regionID -> [opType]Operator
```
Comment (rleungx) [inline server/schedulers/hot_region.go:99]: I think a better comment is to explain how these variables work.

File: server/schedulers/hot_region.go [hunk 3]
```diff
@@ -394,6 +407,7 @@ func filterHotPeers(
 	for _, peer := range peers {
 		if (kind == core.LeaderKind && !peer.IsLeader()) ||
 			peer.HotDegree < minHotDegree ||
+			// hotPeerFilterTy here is always mix currently, isHotPeerFiltered will directly return false here.
```
Comment (lhy1024) [inline server/schedulers/hot_region.go:410]:
```suggestion
			// when hotPeerFilterTy here is always mix currently, isHotPeerFiltered will directly return false here.
```

File: server/schedulers/hot_region.go (hunk 2 above)
Comment (Yisaer) [reply server/schedulers/hot_region.go:99]: updated.

File: server/schedulers/utils.go [hunk 4]
```diff
@@ -197,6 +197,8 @@ func newPendingInfluence(op *operator.Operator, from, to uint64, infl Influence)
 	}
 }
 
+// summaryPendingInfluence calculate the summary pending Influence for each store and return storeID-> Influence
+// It make each rate(key/byte/count) become (1+w) times to the origin value while f is the function to provide w(weight)
```
Comment (rleungx) [inline server/schedulers/utils.go:201]:
```suggestion
// It makes each rate(key/byte/count) become (1+w) times to the origin value while f is the function to provide w(weight)
```

And there is no count rate.

File: server/schedulers/utils.go [hunk 5]
```diff
@@ -197,6 +197,8 @@ func newPendingInfluence(op *operator.Operator, from, to uint64, infl Influence)
 	}
 }
 
+// summaryPendingInfluence calculate the summary pending Influence for each store and return storeID-> Influence
```
Comment (rleungx) [inline server/schedulers/utils.go:200]:
```suggestion
// summaryPendingInfluence calculate the summary pending Influence for each store and return storeID -> Influence
```

File: server/schedulers/hot_region.go [hunk 6]
```diff
@@ -842,6 +869,8 @@ func (bs *balanceSolver) calcProgressiveRank() {
 			}
 			return a - b
 		}
+		// we use DecRatio(Decline Ration) to expect that the dst store's (key/byte) rate should still be less
```
Comment (rleungx) [inline server/schedulers/hot_region.go:872]:
```suggestion
		// we use DecRatio(Decline Ratio) to expect that the dst store's (key/byte) rate should still be less
```

**Comment by disksing** [comment]
It improves understandability a lot!

**Pull request merged**